lib/services/workflow-api-service.js

```diff
@@ -96,11 +96,8 @@ function workflowApiServiceFactory(
                         definition, configuration.options, context, configuration.domain, true);
             });
         })
-        .tap(function(graph) {
-            var progress = GraphProgress.create(graph, 'Graph "' + graph.name + '" started');
-            return eventsProtocol.publishProgressEvent(graph.instanceId,
-                                                       progress.getProgressEventData());
-        })
+        .tap(taskMessenger.publishGraphStarted)
+        .tap(graphProgressService.publishGraphStarted)
```

Comment:
Good catch!
I just refactored the code here but didn't check whether it's the best place to publish the events. We can raise another PR to do that.